### PR TITLE
[Tooling/Inclusion] Update std symbol mapping

### PR DESCRIPTION
**Pull request by vvd170501**
This adds new symbols to the generated mapping and removes special mappings for missing symbols introduced in #113612, as these symbols are now included in the generated mapping.

### CODE REVIEW

File: clang/lib/Tooling/Inclusions/Stdlib/StdSymbolMap.inc [hunk 1]
```diff
@@ -1241,8 +1254,7 @@ SYMBOL(fgetwc, None, <wchar.h>)
 SYMBOL(fgetws, std::, <cwchar>)
 SYMBOL(fgetws, None, <cwchar>)
 SYMBOL(fgetws, None, <wchar.h>)
-SYMBOL(filebuf, std::, <streambuf>)
-SYMBOL(filebuf, std::, <iostream>)
+SYMBOL(filebuf, std::, <fstream>)
```
Comment (vvd170501) [inline clang/lib/Tooling/Inclusions/Stdlib/StdSymbolMap.inc:1257]: Previous versions of cppreference mentioned incorrect headers for `filebuf` and `wfilebuf`.
`<fstream>` is the correct one: https://eel.is/c++draft/fstream.syn#lib:filebuf

**Comment by vvd170501** [comment]
Ping. Could someone review this, please?

**Comment by vvd170501** [comment]
@kadircet, could you review this PR or assign someone else?

**Review by hokein — APPROVED** [review]
Thanks, this looks good.

**Comment by vvd170501** [comment]
@hokein, can you commit this, please? I don't have the permissions

**Comment by hokein** [comment]
> @hokein, can you commit this, please? I don't have the permissions

sure, done.

**Pull request merged**